### PR TITLE
Feature/habit notes

### PR DESCRIPTION
**Pull request by GregVaz**
## Explain why this change is being made.
As a registered user with active habits, I want to see the notes of a specific day of the habit, so that I can see all the notes about the habit.

## Explain how this is accomplished.
Once a habit is created, you could see the list of the days duration of the habit, you could press over each day to add, edit o remove a note

## How do you manually test this?
  - Given the user is logged in, you can press the link Habits over the navbar, then click on 'Add new habit', fill all the form, an once the habit is created you will be redirected to the details page of the habit
  - Once you see the habit details page, if you ever had a note this will show a little blue button, also the calendar will show the current date with a circle painted on the day. 
  - Press the day, and if you do not have a note, then you could add a note, if you already have a note, then you could edit or delete the note

### Screenshots
image
image
image
image
image
image

### CODE REVIEW

File: app/queries/habits_query.rb [hunk 1]
```diff
@@ -12,4 +12,8 @@ def active
   def archived
     @habits.where(status: true)
   end
+
+  def today
+    @habits.where()
```
Comment (jvargas98) [inline app/queries/habits_query.rb]: just a doubt, why is it empty?

Comment (GregVaz) [reply app/queries/habits_query.rb]: I'm sorry, it seems like when I was trying to implement that query I change the method for Days. I'll check that, Thanks Vargas

File: app/assets/stylesheets/note.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the note controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (OmauryMarmolejo) [inline app/assets/stylesheets/note.scss]: Is this file needed?

File: app/controllers/habits_controller.rb [hunk 3]
```diff
@@ -18,10 +17,17 @@ def create
...
+  def show
     @habit = Habit.find(params[:id])
+    @note = Note.new
+    @day_habit = day_habit(@habit)
+    @day_note = @day_habit.note
+  end
+
+  def day_habit(habit)
+    days = DaysQuery.new(@habit).today
+    day_habit = days[0] if days[0]
```
Comment (OmauryMarmolejo) [inline app/controllers/habits_controller.rb]: What about returning this directly on the query

File: app/assets/stylesheets/note.scss (hunk 2 above)
Comment (GregVaz) [reply app/assets/stylesheets/note.scss]: No, it is not yet haha. Thanks

File: app/controllers/habits_controller.rb (hunk 3 above)
Comment (GregVaz) [reply app/controllers/habits_controller.rb]: Yeah it works returning directly the query, thanks!

**Review by jvargas98 — APPROVED** [review]

**Review by ese-varo — APPROVED** [review]
LGTM! :rocket:

File: app/controllers/note_controller.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
+class NoteController < ApplicationController
+  def create
```
Comment (jvargas98) [inline app/controllers/note_controller.rb:2]:
```suggestion
class NoteController < ApplicationController
 before_action :authenticate_user!
 
  def create
```

**Review by jvargas98 — COMMENTED** [review]
I forgot this comment about authentication.

**Review by OmauryMarmolejo — APPROVED** [review]
Cool, just please attend to the comment Jorge left.

File: app/controllers/note_controller.rb (hunk 4 above)
Comment (GregVaz) [reply app/controllers/note_controller.rb:2]: You are completely right, thanks!

**Review by LoginALee — APPROVED** [review]
Great job Goyo :100:
